### PR TITLE
feat: [PL-23301]: browser router

### PR DESCRIPTION
**Pull request by swarajcheguri-harness**
### Summary
**Replace Hash Router with Browser Router and remove # from the URL**
This PR is for testing the changes in QA  once this changes goes to QA. And is under feature flag kind thing which is controlled as of now using Params from the url which will be later replaced with env variables using entrypoint.sh in the coming PR’s

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 7 Issues 
* Code issue 7 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11750)

File: src/framework/app/App.tsx [hunk 1]
```diff
@@ -252,6 +252,14 @@ export function AppWithAuthentication(props: AppProps): React.ReactElement {
 }
 
 export function AppWithoutAuthentication(props: AppProps): React.ReactElement {
+  const { pathname, hash } = window.location
+
+  // Redirect from `/#/account/...` to `/account/...`
+  if (hash && (pathname === '/' || pathname === '/ng')) {
+    const targetUrl = window.location.href.replace('/#/', '/')
```
Comment (knagurski) [inline src/framework/app/App.tsx:259]: Will kill off using a `#` in child apps? I think is something that @tan-nhu needs to be able to do.

File: src/index.html [hunk 2]
```diff
@@ -49,6 +54,9 @@
     <!-- adding bugsnag for SAAS build -->
     <% if (!__DEV__) { %>
     <script>
+      window.NON_CDN_BASE_PATH = `${
+        window.location.pathname.split('/ng').length > 1 ? window.location.pathname.split('/ng')[0] : ''
```
Comment (swarajcheguri-harness) [inline src/index.html:58]: add /ng/ instead of /ng

File: src/framework/app/bootstrap.tsx [hunk 3]
```diff
@@ -7,15 +7,32 @@
...
 const ignoredErrorClasses = ['YAMLSemanticError', 'YAMLSyntaxError', 'AbortError']
-
+const DefaultRouter: React.FC<React.PropsWithChildren<unknown>> = props => {
+  const { browserRouterEnabled } = window
+  return (
+    <>
+      {browserRouterEnabled ? (
+        <BrowserRouter
+          basename={`${
+            window.location.pathname.split('/ng').length > 1 ? window.location.pathname.split('/ng')[0] : ''
```
Comment (swarajcheguri-harness) [inline src/framework/app/bootstrap.tsx:25]: reuse   window.location.pathname.split('/ng').length > 1 ? window.location.pathname.split('/ng' from index .html

File: src/framework/app/App.tsx (hunk 1 above)
Comment (abhinavrastogi-harness) [reply src/framework/app/App.tsx:213]: yes, that's the intention

File: src/index.html [hunk 4]
```diff
@@ -76,6 +87,7 @@
       window.helpPanelAccessToken = '<%= HELP_PANEL_ACCESS_TOKEN_DEV %>'
       window.helpPanelSpace = '<%= HELP_PANEL_SPACE_DEV %>'
       window.helpPanelEnvironment = '<%= HELP_PANEL_ENVIRONMENT_DEV %>'
+      window.NON_CDN_BASE_PATH = '/'
```
Comment (abhinavrastogi-harness) [inline src/index.html:90]: this is defined twice.
repeated from line 65?

File: src/framework/logging/logging.ts [hunk 5]
```diff
@@ -9,7 +9,9 @@ import type { ModuleName } from 'framework/types/ModuleName'
 
 declare global {
   interface Window {
-    bugsnagClient: any // eslint-disable-line
+    bugsnagClient: any // eslint-disable-line,
+    browserRouterEnabled: any //eslint-disable-line
```
Comment (vkbansal-harness) [inline src/framework/logging/logging.ts:13]:
```suggestion
    browserRouterEnabled: boolean
```

File: src/framework/logging/logging.ts [hunk 6]
```diff
@@ -9,7 +9,9 @@ import type { ModuleName } from 'framework/types/ModuleName'
...
+    nameSpace: any //eslint-disable-line
```
Comment (vkbansal-harness) [inline src/framework/logging/logging.ts:14]:
```suggestion
    nameSpace: string
```

File: src/index.html (hunk 4 above)
Comment (swarajcheguri-harness) [reply src/index.html:90]: line 65 is for the non dev env
this is for the dev env

File: src/framework/logging/logging.ts [hunk 7]
```diff
@@ -9,7 +9,9 @@ import type { ModuleName } from 'framework/types/ModuleName'
 
 declare global {
   interface Window {
-    bugsnagClient: any // eslint-disable-line
+    bugsnagClient: any // eslint-disable-line,
+    browserRouterEnabled?: boolean //eslint-disable-line
+    harnessNameSpace: string //eslint-disable-line
```
Comment (abhinavrastogi-harness) [inline src/framework/logging/logging.ts:14]: no need for `eslint-disable-line` since we are not using `any`

**Review by abhinavrastogi-harness — APPROVED** [review]

File: docker/entrypoint.sh [hunk 8]
```diff
@@ -30,6 +31,7 @@ sed -i "s|<\!-- helpPanelEnvironment -->|<script>window.helpPanelEnvironment = '
 sed -i "s|<\!-- newNavContentfulAccessToken -->|<script>window.newNavContentfulAccessToken = '$NEW_NAV_CONTENTFUL_ACCESS_TOKEN'</script>|" index.html
 sed -i "s|<\!-- newNavContetfulSpace -->|<script>window.newNavContetfulSpace = '$NEW_NAV_CONTENTFUL_SPACE'</script>|" index.html
 sed -i "s|<\!-- newNavContentfulEnvironment -->|<script>window.newNavContentfulEnvironment = '$NEW_NAV_CONTENTFUL_ENVIRONMENT'</script>|" index.html
+sed -i "s|<\!-- harnessNameSpacePlaceHolder -->|<script>window.harnessNameSpace = '$HARNESS_NAME_SPACE'</script>|" index.html
```
Comment (mayankVermaHarness) [inline docker/entrypoint.sh:34]: Hadn't we decided to pick up the current name space using `window.location`? Did we change that plan?

If yes, where does `$HARNESS_NAME_SPACE` come from?

Comment (swarajcheguri-harness) [reply docker/entrypoint.sh:34]: Yes i did that using location as we disused
But Abhinav suggested using namespace which was already present in env variables
We will use it here passing it from entrypoint.sh which in turn gets it from manifest files

**Review by vkbansal-harness — DISMISSED** [review]

**Review by mayankVermaHarness — DISMISSED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**